libnmstate/nm/dns.py

```diff
@@ -14,21 +14,32 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
+import six
+
 from itertools import chain
 from operator import itemgetter
 
 from libnmstate import iplib
 from libnmstate.error import NmstateInternalError
 from libnmstate.nm import nmclient
 from libnmstate.nm import active_connection as nm_ac
+from libnmstate.nm.route import IPV4_DEFAULT_GATEWAY_DESTINATION
+from libnmstate.nm.route import IPV6_DEFAULT_GATEWAY_DESTINATION
 from libnmstate.schema import DNS
+from libnmstate.schema import Interface
+from libnmstate.schema import Route
 
 
 DNS_DEFAULT_PRIORITY_VPN = 50
 DNS_DEFAULT_PRIORITY_OTHER = 100
 DNS_METADATA = '_dns'
 DNS_METADATA_PRIORITY = '_priority'
 DEFAULT_DNS_PRIORITY = 0
+# The 40 is chose as default DNS priority of VPN is 50.
+DNS_ORDERING_IPV4_PRIORITY = 40
+DNS_ORDERING_IPV6_PRIORITY = 41
+# Ensuring static DNS config takes priority over DHCP/autoconf
```

Comment:
Ensure

But why will it take priority over DHCP/autoconf, what are their priorities?

Reply:
When DNS priority set to 0(default), the DHCP/Autoconf DNS priority is VPN = 50, other = 100.

Reply:
I assume when user define a static DNS name server, he/she might want them listed before DHCP ones.

Reply:
Thanks for the info. Can you add all this info to the commit message? It will be useful to understand the "why" part for the future us.